source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -0,0 +1,354 @@
+    if (pair.first == Http::Headers::get().AcceptEncodingValues.Identity) {
+      if (pair.second > 0) {
+        config_->stats().header_identity_.inc();
+      } else {
+        config_->stats().header_not_valid_.inc();
+      }
+      return false;
+    }
+
+    // If wildcard is given then use which ever compressor is registered first.
+    if (pair.first == Http::Headers::get().AcceptEncodingValues.Wildcard) {
+      if (pair.second > 0 && !allowed_compressors.empty()) {
+        config_->stats().header_wildcard_.inc();
+        auto first_registered = std::min_element(
+            allowed_compressors.begin(), allowed_compressors.end(),
+            [](const std::pair<std::string, uint32_t>& a,
+               const std::pair<std::string, uint32_t>& b) -> bool { return a.second < b.second; });
+        return StringUtil::caseCompare(config_->contentEncoding(), first_registered->first);
+      } else {
+        config_->stats().header_not_valid_.inc();
+        return false;
+      }
+    }
+  }
+
+  config_->stats().header_not_valid_.inc();
+  return false;
+}
+
+bool CompressorFilter::isContentTypeAllowed(Http::HeaderMap& headers) const {
+  const Http::HeaderEntry* content_type = headers.ContentType();
+  if (content_type && !config_->contentTypeValues().empty()) {
+    const absl::string_view value =
+        StringUtil::trim(StringUtil::cropRight(content_type->value().getStringView(), ";"));
+    return config_->contentTypeValues().find(value) != config_->contentTypeValues().end();
+  }
+
+  return true;
+}
+
+bool CompressorFilter::isEtagAllowed(Http::HeaderMap& headers) const {
+  const bool is_etag_allowed = !(config_->disableOnEtagHeader() && headers.Etag());
+  if (!is_etag_allowed) {
+    config_->stats().not_compressed_etag_.inc();
+  }
+  return is_etag_allowed;
+}
+
+bool CompressorFilter::isMinimumContentLength(Http::HeaderMap& headers) const {
+  const Http::HeaderEntry* content_length = headers.ContentLength();
+  if (content_length) {
+    uint64_t length;
+    const bool is_minimum_content_length =
+        absl::SimpleAtoi(content_length->value().getStringView(), &length) &&
+        length >= config_->minimumLength();
+    if (!is_minimum_content_length) {
+      config_->stats().content_length_too_small_.inc();
+    }
+    return is_minimum_content_length;
+  }
+
+  const Http::HeaderEntry* transfer_encoding = headers.TransferEncoding();
+  return (transfer_encoding &&
+          StringUtil::caseFindToken(transfer_encoding->value().getStringView(), ",",
+                                    Http::Headers::get().TransferEncodingValues.Chunked));
+}
+
+bool CompressorFilter::isTransferEncodingAllowed(Http::HeaderMap& headers) const {
+  const Http::HeaderEntry* transfer_encoding = headers.TransferEncoding();
+  if (transfer_encoding) {
+    for (auto header_value :
+         // TODO(gsagula): add Http::HeaderMap::string_view() so string length doesn't need to be
```

Comment:
I think this TODO can be removed due to a previous refactor to use getStringView().